src/sentry/api/endpoints/release_deploys.py

```diff
@@ -119,6 +119,10 @@ def post(self, request, organization, version):
                 last_deploy_id=deploy.id,
             )
 
+            ReleaseProjectEnvironment.objects.filter(release=release, environment=env).update(
```

Comment:
You also need project in this filter or it'll be a bad time for the index.

Reply:
mm would like `project__in=release.projects.all()` be ok?

Reply:
You already have `projects` above as a list, without it being a queryset, but yeah.